### PR TITLE
Fixed code smell issues on main branch, changed the IEventProxy use i…

### PR DESCRIPTION
**Pull request by mhawkinsbasis**
Fixed code smell issues on main branch, changed the IEventProxy use in BBjGridExWidgetSelectEventProxy, and renamed ButtonPushEvent to ButtonClickEvent (updated imports as well)

### CODE REVIEW

**Comment by hyyan** [comment]
We should try reduce duplications before merging. duplications went from 4.0% to 20.1% with the PR

**Comment by mhawkinsbasis** [comment]
Closes #46 #47 #52

**Review by hyyan — CHANGES_REQUESTED** [review]

File: src/main/java/org/dwcj/controls/listbox/ListBox.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import com.basis.bbj.proxies.sysgui.BBjWindow;
 import com.basis.startup.type.BBjException;
 
+import edu.emory.mathcs.backport.java.util.Collections;
```
[the remark this answers is not in the document]
Comment (mhawkinsbasis) [reply src/main/java/org/dwcj/controls/listbox/ListBox.java:7]: This is being used in a few of the Listbox's methods that need to return an empty map i.e. getSelectedItems() which can return a Collections.emptyMap();

File: src/main/java/org/dwcj/controls/panels/Div.java [hunk 2]
```diff
@@ -107,17 +107,18 @@ public Div setStyle(String property, String value) {
     }
     
     @Override
-    public Div addClass(String selector) {
-        super.addClass(selector);
+    public Div addClassName(String selector) {
```
[the remark this answers is not in the document]
Comment (hyyan) [reply src/main/java/org/dwcj/controls/panels/Div.java:110]: We're going to address this later

**Review by hyyan — APPROVED** [review]

File: src/main/java/org/dwcj/controls/AbstractControl.java [hunk 3]
```diff
@@ -114,8 +114,8 @@ public Boolean getCaughtUp() {
     @SuppressWarnings("java:S3776") // tolerate cognitive complexity for now, it's just a batch list of checks
     protected void catchUp() throws IllegalAccessException {
 
-        if (this.caughtUp)
-            throw new IllegalAccessException("catchUp cannot be called twice");
+        if (Boolean.TRUE.equals(Boolean.TRUE.equals(this.caughtUp))
```
[the remark this answers is not in the document]
Comment (hyyan) [reply src/main/java/org/dwcj/controls/AbstractControl.java]: @see https://github.com/DwcJava/engine/issues/59

**Pull request merged**